js/main.js

```diff
@@ -3,260 +3,255 @@
-    fMapping.add(g_workspace, 'scaleId', {'Linear': Workspace.Scale.LINEAR.id, 'Logarithmic': Workspace.Scale.LOG.id}).name('Scale');
-    fMapping.add(g_workspace, 'hotspotQuantile').name('Hotspot quantile').step(0.0001);
-    var colorMaps = Object.keys(ColorMap.Maps).reduce(function(m, k) {
-        m[ColorMap.Maps[k].name] = k;
-        return m;
-    }, {});
-    fMapping.add(g_workspace, 'colorMapId', colorMaps).name('Color map');
-
-    var mapping = {
-        flag: fMapping.add(g_workspace, 'autoMinMax').name('Auto MinMax'),
-        min: fMapping.add(g_workspace, 'minValue').name('Min value').step(0.00001),
-        max: fMapping.add(g_workspace, 'maxValue').name('Max value').step(0.00001),
-    };
-    g_workspace.addEventListener(Workspace.Events.AUTO_MAPPING_CHANGE,
-                                 onAutoMappingChange.bind(null, mapping));
-    onAutoMappingChange(mapping);
+    /*
+     * Initializing DAT.GUI (http://workshop.chromeexperiments.com/examples/gui) controls.
+     */
+    function initGUI() {
+        g_gui = new dat.GUI();
+
+        var f2d = g_gui.addFolder('2D');
+        f2d.add(g_workspace.scene2d, 'spotBorder', 0, 1).name('Spot border').step(0.01);
+
+        var f3d = g_gui.addFolder('3D');
+        f3d.add(g_views.g3d, 'layout', {
+            'Single view': ViewGroup3D.Layout.SINGLE,
+            'Double view': ViewGroup3D.Layout.DOUBLE,
+            'Triple view': ViewGroup3D.Layout.TRIPLE,
+            'Quadriple view': ViewGroup3D.Layout.QUADRIPLE,
+        }).name('Layout');
+        f3d.addColor(g_workspace.scene3d, 'color').name('Color');
+        f3d.addColor(g_workspace.scene3d, 'backgroundColor').name('Background');
+        f3d.add(g_workspace.scene3d.frontLight, 'intensity', 0, 3).name('Light');
+        f3d.add(g_workspace.scene3d, 'spotBorder', 0, 1).name('Spot border').step(0.01);
+        f3d.add(g_views, 'exportPixelRatio3d', [0.5, 1.0, 2.0, 4.0]).name('Export pixel ratio');
+        var adjustment = f3d.addFolder('Adjustment');
+        adjustment.add(g_workspace.scene3d.adjustment, 'alpha', -180.0, 180.0).name('0X rotation').step(1);
+        adjustment.add(g_workspace.scene3d.adjustment, 'beta', -180.0, 180.0).name('0Y rotation').step(1);
+        adjustment.add(g_workspace.scene3d.adjustment, 'gamma', -180.0, 180.0).name('0Z rotation').step(1);
+        adjustment.add(g_workspace.scene3d.adjustment, 'x').name('X offset').step(0.1);
+        adjustment.add(g_workspace.scene3d.adjustment, 'y').name('Y offset').step(0.1);
+        adjustment.add(g_workspace.scene3d.adjustment, 'z').name('Z offset').step(0.1);
+
+        var fMapping = g_gui.addFolder('Mapping');
+        fMapping.add(g_workspace, 'scaleId', {'Linear': Workspace.Scale.LINEAR.id, 'Logarithmic': Workspace.Scale.LOG.id}).name('Scale');
+        fMapping.add(g_workspace, 'hotspotQuantile').name('Hotspot quantile').step(0.0001);
+        var colorMaps = Object.keys(ColorMap.Maps).reduce(function(m, k) {
```

Comment:
Good catch, fixed. Thanks!

On (Apr-14-16|15:09), Ivan Protsyuk wrote:
> > +        f3d.addColor(g_workspace.scene3d, 'backgroundColor').name('Background');
> > +        f3d.add(g_workspace.scene3d.frontLight, 'intensity', 0, 3).name('Light');
> > +        f3d.add(g_workspace.scene3d, 'spotBorder', 0, 1).name('Spot border').step(0.01);
> > +        f3d.add(g_views, 'exportPixelRatio3d', [0.5, 1.0, 2.0, 4.0]).name('Export pixel ratio');
> > +        var adjustment = f3d.addFolder('Adjustment');
> > +        adjustment.add(g_workspace.scene3d.adjustment, 'alpha', -180.0, 180.0).name('0X rotation').step(1);
> > +        adjustment.add(g_workspace.scene3d.adjustment, 'beta', -180.0, 180.0).name('0Y rotation').step(1);
> > +        adjustment.add(g_workspace.scene3d.adjustment, 'gamma', -180.0, 180.0).name('0Z rotation').step(1);
> > +        adjustment.add(g_workspace.scene3d.adjustment, 'x').name('X offset').step(0.1);
> > +        adjustment.add(g_workspace.scene3d.adjustment, 'y').name('Y offset').step(0.1);
> > +        adjustment.add(g_workspace.scene3d.adjustment, 'z').name('Z offset').step(0.1);
> > +
> > +        var fMapping = g_gui.addFolder('Mapping');
> > +        fMapping.add(g_workspace, 'scaleId', {'Linear': Workspace.Scale.LINEAR.id, 'Logarithmic': Workspace.Scale.LOG.id}).name('Scale');
> > +        fMapping.add(g_workspace, 'hotspotQuantile').name('Hotspot quantile').step(0.0001);
> > +        var colorMaps = Object.keys(ColorMap.Maps).reduce(function(m, k) {
> 
> I guess ColorMar needs to be added as a dependency
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/ili-toolbox/ili/pull/72/files/cd65a391f044a50b8c5f5df042d070f3fb3ab7a4#r59800433